pvlib/inverter.py

```diff
@@ -91,29 +125,74 @@ def sandia(v_dc, p_dc, inverter):
     '''
 
     Paco = inverter['Paco']
-    Pdco = inverter['Pdco']
-    Vdco = inverter['Vdco']
-    Pso = inverter['Pso']
-    C0 = inverter['C0']
-    C1 = inverter['C1']
-    C2 = inverter['C2']
-    C3 = inverter['C3']
     Pnt = inverter['Pnt']
+    Pso = inverter['Pso']
 
-    A = Pdco * (1 + C1 * (v_dc - Vdco))
-    B = Pso * (1 + C2 * (v_dc - Vdco))
-    C = C0 * (1 + C3 * (v_dc - Vdco))
-
-    power_ac = (Paco / (A - B) - C * (A - B)) * (p_dc - B) + C * (p_dc - B)**2
-    power_ac = np.minimum(Paco, power_ac)
-    power_ac = np.where(p_dc < Pso, -1.0 * abs(Pnt), power_ac)
+    power_ac = _sandia_eff(v_dc, p_dc, inverter)
+    power_ac = _sandia_limits(power_ac, p_dc, Paco, Pnt, Pso)
 
     if isinstance(p_dc, pd.Series):
         power_ac = pd.Series(power_ac, index=p_dc.index)
 
     return power_ac
 
 
+def sandia_multi(v_dc, p_dc, inverter):
+    r'''
+    Convert DC power and voltage to AC power for an inverter with multiple
+    MPPT inputs.
+
+    Uses Sandia's Grid-Connected PV Inverter model [1]_.
+
+    Parameters
+    ----------
+    v_dc : tuple or list of numeric
+        DC voltage on each MPPT input of the inverter. [V]
+
+    p_dc : tuple or list of numeric
+        DC power on each MPPT input of the inverter. [W]
+
+    inverter : dict-like
+        Defines parameters for the inverter model in [1]_.
+
+    Returns
+    -------
+    power_ac : numeric
+        AC power output for the inverter. [W]
+
+    Raises
+    ------
+    ValueError if v_dc and p_dc have different lengths.
+
+    Notes
+    -----
+    See :py:func:`pvlib.inverter.sandia` for definition of the parameters in
+    `inverter`.
+
+    References
+    ----------
+    .. [1] D. King, S. Gonzalez, G. Galbraith, W. Boyson, "Performance Model
+       for Grid-Connected Photovoltaic Inverters", SAND2007-5036, Sandia
+       National Laboratories.
+
+    See also
+    --------
+    pvlib.inverter.sandia
+    '''
+
+    if len(p_dc) == len(v_dc):
```

Comment:
change to `!=` and put the `raise` in the `if` block, then move the rest of the code out of the `if/else` structure. no need for an `else`.